### PR TITLE
Re-Add Block Validity Checks

### PR DESCRIPTION
**Pull request by rauljordan**
Hi all,

This re-adds block validity check which was accidentally removed in a previous PR to the block processing in our beacon-chain/blockchain/service.go file.

### CODE REVIEW

File: beacon-chain/blockchain/service.go [hunk 1]
```diff
@@ -340,7 +341,29 @@ func (c *ChainService) blockProcessing() {
...
+			if isTransition {
+				cState, err = cState.NewStateRecalculations(aState, block)
+				if err != nil {
+					log.Errorf("Initialize new cycle transition failed: %v", err)
+					continue
+				}
+			}
+
+			if valid := block.IsValid(aState, cState, parent.SlotNumber()); !valid {
+				log.Debugf("Block failed validity conditions: %v", err)
```
Comment (terenc3t) [inline beacon-chain/blockchain/service.go]: if a block fails validity, should it be debug or error? I think error is more appropriate here

Comment (rauljordan) [reply beacon-chain/blockchain/service.go]: Debug - we'll see tons of blocks failing validity conditions from bad actors or bad proposers

File: beacon-chain/blockchain/service.go [hunk 2]
```diff
@@ -260,9 +263,11 @@ func (c *ChainService) updateHead(slotInterval <-chan time.Time) {
 				continue
 			}
 
-			if err := c.chain.SetCrystallizedState(cState); err != nil {
-				log.Errorf("Write crystallized state to disk failed: %v", err)
-				continue
+			if stateTransitioned {
```
Comment (rawfalafel) [inline beacon-chain/blockchain/service.go]: Note: this addresses #460

File: beacon-chain/blockchain/service.go [hunk 3]
```diff
@@ -232,23 +232,26 @@ func (c *ChainService) updateHead(slotInterval <-chan time.Time) {
...
+			var stateTransitioned bool
 
-			if isTransition {
+			for cState.IsCycleTransition(parentBlock.SlotNumber()) {
 				cState, err = cState.NewStateRecalculations(aState, block)
 				if err != nil {
 					log.Errorf("Initialize new cycle transition failed: %v", err)
 					continue
 				}
+				stateTransitioned = true
```
Comment (rawfalafel) [inline beacon-chain/blockchain/service.go]: This variable and `isTransition` is redundant. I'd suggest just sticking to `isTransition`.

Comment (rawfalafel) [reply beacon-chain/blockchain/service.go]: whoops, ignore this!

File: beacon-chain/blockchain/core.go [hunk 4]
```diff
@@ -169,6 +169,10 @@ func (b *BeaconChain) PersistCrystallizedState() error {
 	return b.db.Put(crystallizedStateLookupKey, encodedState)
 }
 
+func (b *BeaconChain) saveCrystallizedStateForBlock(state *types.CrystallizedState, block *types.Block) error {
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/core.go]: What is this? It is not used in your PR and does nothing.

Please revert it

**Pull request merged**